### PR TITLE
Add CSRF filtering

### PR DESCRIPTION
**Pull request by davidfurey**
## Why are you doing this?

To protect against cross site request forging attacks.

## Changes

* Enable CSRF on private actions
* Check that cookies haven't been read or set for cached pages

### CODE REVIEW

File: assets/helpers/csrf/csrfReducer.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+// ----- Types ----- //
+
+type Csrf = {
+    token: ?string,
+};
+
+
+// ----- Setup ----- //
+
+const initialState: Csrf = window.guardian && window.guardian.csrf ? {
```
Comment (svillafe) [inline assets/helpers/csrf/csrfReducer.js]: As a good practice, it is better to keep the initial state constant, i.e. not depending on the window object and pass the initial state when you create it. The function `[createStore](http://redux.js.org/docs/api/createStore.html)` receives the initial state as a parameter. In addition, if the actions do not depend on the window object, it is easier for us to test them with Jest. See [here](https://github.com/guardian/support-frontend/blob/master/assets/pages/bundles-landing/bundlesLanding.jsx#L25) for an example of this.

Comment (Ap0c) [inline assets/helpers/csrf/csrfReducer.js]: Hmm, @davidfurey were you basing this on what I did for the [user reducer](https://github.com/guardian/support-frontend/blob/master/assets/helpers/user/userReducer.js#L21)? Looking at what I did there I think I agree with @svillafe, i.e. that wasn't the best solution.

What I should have done is passed this through to the `createStore` function as the `preloadedState`. The [docs](http://redux.js.org/docs/recipes/reducers/InitializingState.html) describe this parameter as being used for 'hydrating the store from some persistent storage or the server', which I think is what we're doing both here and in the case of the user reducer.

File: assets/helpers/csrf/csrfActions.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+
+// ----- Actions Creators ----- //
+
+export function setToken(name: string): Action {
+  return { type: 'SET_TOKEN', name };
+}
+
+export default setToken;
+
```
Comment (Ap0c) [inline assets/helpers/csrf/csrfActions.js]: Will this action ever get used? If not, we may not need it; this state will just remain static client-side after being hydrated from the server.

File: assets/helpers/csrf/csrfReducer.js (hunk 1 above)
Comment (svillafe) [inline assets/helpers/csrf/csrfReducer.js]: @Ap0c Yeap, we should extract the window from the user reducer. I created a [card](https://trello.com/c/Z3Mjst6V/714-refactor-extract-the-window-object-from-the-userreducer-and-add-jest-test) to address this refactor.

Comment (davidfurey) [inline assets/helpers/csrf/csrfReducer.js]: This is entirely based on the user reducer.  Should we fix them both at the same time?

Comment (svillafe) [inline assets/helpers/csrf/csrfReducer.js]: From my point of view, I guess that is better to keep this PR focus on CSRF and refactor UserReducer in a separate PR. Additionally, we can add tests to that reducer in that PR.

Comment (davidfurey) [inline assets/helpers/csrf/csrfReducer.js]: I didn't mean fix the user reducer in this PR, I meant fix them both in a new PR :-)

Maybe on Monday one of you can explain the `preloadedState` approach?

Comment (svillafe) [inline assets/helpers/csrf/csrfReducer.js]: hahah ok, Fair enough, I don't mind to refactor everything next week.

**Pull request merged**